core/http-auth-spi/src/main/java/software/amazon/awssdk/http/auth/spi/NoAuthAuthScheme.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.http.auth.spi;
+
+import software.amazon.awssdk.annotations.SdkPublicApi;
+import software.amazon.awssdk.http.auth.spi.internal.DefaultNoAuthAuthScheme;
+import software.amazon.awssdk.identity.spi.Identity;
+
+/**
+ * An auth scheme that represents no authentication.
+ */
+@SdkPublicApi
+public interface NoAuthAuthScheme extends AuthScheme<NoAuthAuthScheme.AnonymousIdentity> {
+    /**
+     * The scheme ID for the no-auth auth scheme.
+     */
+    String SCHEME_ID = "smithy.api#noAuth";
+
+    static NoAuthAuthScheme create() {
+        return DefaultNoAuthAuthScheme.create();
+    }
+
+    /**
+     * An anonymous identity used by the no-auth auth scheme.
+     */
+    interface AnonymousIdentity extends Identity {
```

Comment:
I think we could define this directly in identity-spi, but we can leave that for surface area review if needed.

Reply:
can we add a TODO comment for this?